presets/tuning/text-generation/dataset.py

```diff
@@ -0,0 +1,154 @@
+            self.dataset = load_dataset(file_ext, data_files=dataset_path, split="train")
+            print(f"Dataset loaded successfully from {dataset_path} with file type '{file_ext}'.")
+        except Exception as e:
+            print(f"Error loading dataset: {e}")
+            raise ValueError(f"Unable to load dataset {dataset_path} with file type '{file_ext}'")
+
+    def find_valid_dataset(self, data_dir):
+        """ Searches for a file with a valid dataset type in the given directory. """
+        for root, dirs, files in os.walk(data_dir):
+            for file in files:
+                filename_lower = file.lower()  # Convert to lowercase once per filename
+                for ext in SUPPORTED_EXTENSIONS:
+                    if ext in filename_lower:
+                        return os.path.join(root, file)
+        return None
+
+    def get_file_extension(self, file_path):
+        """ Returns the file extension based on filetype guess or filename. """
+        filename_lower = os.path.basename(file_path).lower()
+        for ext in SUPPORTED_EXTENSIONS:
+            if ext in filename_lower:
+                return ext
+        _, file_ext = os.path.splitext(file_path)
+        return file_ext[1:]  # Remove leading "."
+
+    def shuffle_dataset(self, seed=None):
+        if self.dataset is None:
+            raise ValueError("Dataset is not loaded.")
+        self.dataset = self.dataset.shuffle(seed=seed)
+
+    def split_dataset(self):
+        if self.dataset is None:
+            raise ValueError("Dataset is not loaded.")
+        assert 0 < self.config.train_test_split <= 1, "Train/Test split needs to be between 0 and 1"
+        if self.config.train_test_split < 1:
+            split_dataset = self.dataset.train_test_split(
+                test_size=1-self.config.train_test_split,
+                seed=self.config.shuffle_seed
+            )
+            return split_dataset['train'], split_dataset['test']
+        else:
+            return self.dataset, None
+
+    def get_dataset(self):
+        if self.dataset is None:
+            raise ValueError("Dataset is not loaded.")
+        return self.dataset
+
+    def format_and_preprocess(self):
+        # OAI Compliant: https://platform.openai.com/docs/guides/fine-tuning/preparing-your-dataset
+        # https://github.com/huggingface/trl/blob/main/trl/extras/dataset_formatting.py
+        # https://huggingface.co/docs/trl/en/sft_trainer#dataset-format-support
+        if self.config.messages_column:
+            self.format_conversational()
+        elif self.config.context_column and self.config.response_column:
+            self.format_instruct()
+        elif self.config.response_column:
+            self.dataset = self.dataset.map(
+                lambda example: self.tokenizer(example[self.config.response_column], **self.tokenizer_params),
+                batched=True
+            )
+            self.format_text()
+            return self.config.response_column
```

Comment:
Just curious: Why we return the response_column here but otherwise return none?

Reply:
fixed using instance variable